packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { store as blocksStore } from '@wordpress/blocks';
+import {
+	PanelBody,
+	__experimentalItemGroup as ItemGroup,
+	__experimentalItem as Item,
+} from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../lock-unlock';
+import InspectorControls from '../components/inspector-controls';
+import {
+	blockBindingsKey,
+	useBlockEditContext,
+} from '../components/block-edit/context';
+
+export const BlockBindingsPanel = () => {
+	const blockEditContext = useBlockEditContext();
+	const bindings = blockEditContext[ blockBindingsKey ];
+
+	const { sources } = useSelect( ( select ) => {
+		const _sources = unlock(
+			select( blocksStore )
+		).getAllBlockBindingsSources();
+
+		return {
+			sources: _sources,
+		};
+	}, [] );
+
+	if ( ! bindings ) {
+		return null;
+	}
+
+	return (
+		<InspectorControls>
+			<PanelBody
+				title={ __( 'Bindings' ) }
+				className="components-panel__block-bindings-panel"
+			>
+				<ItemGroup isBordered isSeparated size="large">
+					{ Object.keys( bindings ).map( ( key, index ) => {
```

Comment:
Nitpick. `key` is the attribute name here, and it's unique so I don't think you need to use `index` at all.